File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task42_PERPLEXITY_llama-3-sonar-large-32k-chat.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class AuthenticationSystem {
    private Map<String, String> users;

    public AuthenticationSystem() {
        this.users = new HashMap<>();
        this.users.put("admin", "password123");
        this.users.put("user", "password456");
    }

    public boolean authenticate(String username, String password) {
        return this.users.getOrDefault(username, "").equals(password);
    }

    public static void main(String[] args) {
        AuthenticationSystem auth = new AuthenticationSystem();
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter username: ");
        String username = scanner.nextLine();

        System.out.print("Enter password: ");
        String password = scanner.nextLine();

        if (auth.authenticate(username, password)) {
            System.out.println("Authentication successful!");
        } else {
            System.out.println("Invalid credentials!");
        }
    }
}

File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task42_PERPLEXITY_llama-3-sonar-large-32k-chat.cpp
code here:
#include <iostream>
#include <string>
#include <vector>

class User {
private:
    std::string username;
    std::string password;

public:
    User(std::string username, std::string password) : username(username), password(password) {}

    bool authenticate(std::string username, std::string password) {
        if (this->username == username && this->password == password) {
            return true;
        } else {
            return false;
        }
    }
};

int main() {
    std::vector<User> users = {{User("user1", "password1"), User("user2", "password2")}};
    std::string username, password;
    std::cout << "Enter username: ";
    std::cin >> username;
    std::cout << "Enter password: ";
    std::cin >> password;
    for (const auto& user : users) {
        if (user.authenticate(username, password)) {
            std::cout << "Authentication successful!" << std::endl;
            return 0;
        }
    }
    std::cout << "Invalid credentials!" << std::endl;
    return 0;
}